File: scripts/maui-vml-verify/corpus/fortran/skel.f90
! skel.f90 —— Fortran 前端「能不能写游戏」最小骨架（期望输出恰好一行 SKEL-SUM=14）
!
! 取材 Examples/fortran/math/main.f90（`program … implicit none … end program`）与
! Examples/fortran/bench.f90（`do i = 1, 4 … end do`）；前端实现 VMLPrepares/FortranCompiler/。
!
! 已修（原本跑不过，现在这两条都是前端的回归点）：
!  ① 数组**读**：表达式里的 `a(i)` 原先被解析成 FuncCallNode → `CALL func_a`
!     → 链接期「未找到标签: func_a」、运行期 KeyNotFound。同款缺陷在**上游真例子**
!     Examples/fortran/sorting/main.f90（`arr(j)`）上同样复现 ⇒ 是前端缺陷，不是语料写错。
!     现在 Parser 按「声明成了数组」判成 ArrayElemNode（见 ArrayElemNode / IsDeclaredArray），
!     并且 `integer :: a(4)` 会真的留出 4×4 字节栈帧（GenerateVarDecl → GetVarByteSize）。
!  ② `call ui_rect(...)` 会被编成 `CALL sub_ui_rect`，而链接器只剥
!     func_/word_/method_/var_ 四种前缀（LibraryLinker.cs:162-182），**不含 sub_**。
!     所以这里改用函数调用表达式 `k = ui_rect(...)` → `CALL func_ui_rect` → 可解析到
!     lib_vmlui_ui_rect（实参逆序压栈，Expressions.cs:196-205）。这是偏离标准 Fortran
!     的一处，纯粹为了碰得到 UI 库。
!
! 有意偏离（不是缺陷）：
!  ③ 打印**不用** `print *, 'SKEL-SUM=', s`：Fortran 的列表输出在列表项之间本来就插
!     分隔符（correct Fortran 会打出 `SKEL-SUM= 14`），拿不到逐字节的 `SKEL-SUM=14`。
!     这是语料的期望错，不是前端缺陷。照 Go/Python/Kotlin/Java/C# 语料的既有做法：
!     先打一个**不带换行**的字符串（`k = print_str(...)`，k 只是接住返回值），
!     再单独打这个数（单个列表项不插分隔符）⇒ 输出恰好 `SKEL-SUM=14`。
!  ④ Fortran 数组 1-based ⇒ 下标写 1..4，循环 do i = 1, 4。

program skel
  implicit none
  integer :: a(4)
  integer :: i
  integer :: s
  integer :: k

  a(1) = 1
  a(2) = 2
  a(3) = 3
  a(4) = 4
  s = 0
  do i = 1, 4
    a(i) = inc(a(i))
    s = s + a(i)
  end do
  k = print_str('SKEL-SUM=')
  print *, s
  k = ui_rect(10, 10, 50, 50, -65536, 1, 0, 0)
  k = ui_present()
contains
  integer function inc(x)
    integer :: x
    inc = x + 1
  end function inc
end program skel

File: scripts/maui-vml-verify/corpus/kotlin/skel.kt
// skel.kt —— VML 骨架程序（Kotlin）。
//
// 习惯用法抄自 Examples/kotlin/fibonacci/main.kt、geometry/main.kt（fun 定义；入口是 fun main，
// 见 KotlinCompiler/CodeGenerator.cs:107）。示例文件都是纯函数文件，没有循环/数组可抄。
//
// 外部库函数：KotlinCompiler 对**任何**不认识的函数名都发裸标签 CALL、实参从右到左压栈
// （CodeGenerator.cs:736-762）⇒ 与库包装的 C 约定一致，ui_rect / ui_present 可以直接裸调。
//
// ⚠ 打印不用 println()：println/print 在 Parser.cs:136-149 是**关键字形式**、只吃一个实参，
//   且 println 自带换行 ⇒ 拼不出 "SKEL-SUM=14" 一行。改用库函数 print_str（不换行）
//   + println_int（含换行）。
// ⚠ 不确定：`a[i] = v` 这种**下标赋值**在 Kotlin 的 Parser 里没有对应分支
//   （赋值只认裸名与成员访问），按源码看会解析失败 —— 若真编不过，缺的是前端这个语法，
//   不是本骨架写错。这正是要测的一条，所以照自然写法写、不做规避。
// 数组用 arrayOf(...)（docs/前端游戏能力评估.md §4.2C 用的就是它）；颜色写负数十进制。
fun plus1(x: Int): Int { return x + 1 }

fun main() {
    val a = arrayOf(1, 2, 3, 4)
    var s = 0
    for (i in 0..3) {
        a[i] = plus1(a[i])
        s = s + a[i]
    }
    ui_rect(10, 10, 50, 50, -65536, 1, 0, 0)
    ui_present()
    print_str("SKEL-SUM=")
    println_int(s)
}
